### PR TITLE
switch implemented in mobile

### PR DESCRIPTION
**Pull request by chrisnojima**
calling this checkbox since we have flow for  it already and the api is identical, also takes in a label too since we'll basically always do this

@keybase/react-hackers

### CODE REVIEW

File: shared/common-adapters/checkbox.native.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  props: Props;
+  state: State;
+
+  _getOffset (props: Props): number {
+    return props.checked ? checkedOffset : 0
+  }
+
+  constructor (props: Props) {
+    super(props)
+    this.state = {left: new Animated.Value(this._getOffset(props))}
```
Comment (chrisnojima) [inline shared/common-adapters/checkbox.native.js]: this uses the animated framework, so read up on that if you've never seen this before

File: shared/common-adapters/checkbox.native.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    if (this.props.checked !== nextProps.checked) {
+      Animated.timing(this.state.left, {
+        toValue: this._getOffset(nextProps),
+        easing: Easing.linear
+      }).start()
+    }
+  }
+
+  shouldComponentUpdate (nextProps: Props, nextState: State): boolean {
+    return (
```
Comment (chrisnojima) [inline shared/common-adapters/checkbox.native.js]: we might want to enforce that all common-adapters have shouldComponentUpdate calls, else we'll likely have empty renders

File: shared/more/style-sheet.render.native.js [hunk 3]
```diff
@@ -178,10 +176,40 @@ const Inputs = () => (
   <Text type='Header'>TODO</Text>
 )
 
+const Checkboxes = ({check, flip}) => {
+  return (
+    <Box>
+      {false && <Row><Switch onTintColor={globalColors.blue} value={check[0]} onValueChange={() => flip(0)}/></Row>}
```
Comment (chrisnojima) [inline shared/more/style-sheet.render.native.js]: this false stuff is just so we can compare it vs the native components. might as well leave this

File: shared/more/style-sheet.render.native.js [hunk 4]
```diff
@@ -178,10 +176,40 @@ const Inputs = () => (
...
+  flip (idx: number) {
+    const next = {...this.state.check}
+    next[idx] = !next[idx]
+    this.setState({check: next})
+  }
+
   render () {
     return (
       <ScrollView>
+        <Container title='Checkboxes'><Checkboxes flip={idx => this.flip(idx)} check={this.state.check}/></Container>
```
Comment (chrisnojima) [inline shared/more/style-sheet.render.native.js]: passing this in so we can have it actually flip in the component

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

File: shared/common-adapters/checkbox.native.js (hunk 1 above)
Comment (MarcoPolo) [inline shared/common-adapters/checkbox.native.js]: This is pretty neat!

File: shared/common-adapters/checkbox.native.js (hunk 2 above)
Comment (MarcoPolo) [inline shared/common-adapters/checkbox.native.js]: What do you mean?

**Comment by MarcoPolo** [comment]
:+1: 

just with a question, but you can merge

File: shared/common-adapters/checkbox.native.js (hunk 2 above)
Comment (chrisnojima) [inline shared/common-adapters/checkbox.native.js]: i just noticed wasted render calls where we re-render when we didn't need to. We should likely ensure our commonly used (and used a lot) components don't have those sorts of issues. Not a big deal now but probably worth doing it on these guys

**Pull request merged**